addons/mail/static/src/core/common/chat_window.xml

```diff
@@ -67,6 +67,18 @@
                 <t t-set="action" t-value="threadActions.actions.at(-1)"/>
                 <t t-set="itemClass" t-value="'me-1'"/>
             </t>
+            <t t-if="this.store.inPublicPage">
+                <button class="btn ps-1" t-if="!state.editingGuestName">
+                    <img class="o-mail-Discuss-selfAvatar rounded-circle o_object_fit_cover flex-shrink-0" alt="Avatar" t-att-src="store.self.avatarUrl" t-on-click="() => state.editingGuestName = true"/>
+                </button>
+                <AutoresizeInput
+                    t-if="state.editingGuestName"
+                    className="'py-1 me-2'"
+                    autofocus="true"
+                    onValidate.bind="renameGuest"
+                    value="store.self.name"
+                />
+            </t>
```

Comment:
Only guests can change their name. Internal users cannot (this is linked to their account / partner)

Reply:
The guard `this.store.inPublicPage` should already make sure this is only displayed to guests, does it not?

Reply:
No, you can access the public guest page while being authenticated with an internal user. So `this.store.inPublicPage` is not enough to ensure this is guest-only branching